Document UseHttps limitations for Http/3 in Kestrel (#23430)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/servers/kestrel/http3.md

* `HttpClient` can be used for localhost/loopback testing in .NET 6 or later. Extra configuration is required when using `HttpClient` to make an HTTP/3 request:
  * Set `HttpRequestMessage.Version` to 3.0, or
  * Set `HttpRequestMessage.VersionPolicy` to `HttpVersionPolicy.RequestVersionOrHigher`.

## Limitations

> Certain HTTPS scenarios are not yet supported for HTTP/3 in Kestrel. When calling <xref:Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps> with <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> while using HTTP/3, setting the following options on the <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> will no-op:

Reviewer: ```suggestion
Some HTTPS scenarios are not yet supported for HTTP/3 in Kestrel. When calling `Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps` with <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> while using HTTP/3, setting the following options on the <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> is a no-op (it does nothing):
```

* <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions.HandshakeTimeout>
* <xref:Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions.OnAuthenticate>

> Calling the following implementations of <xref:Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps> will throw an error when using HTTP/3:

Reviewer: ```suggestion
Calling the following implementations of `Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps` throw an error when using HTTP/3:
```

> `UseHttps(this ListenOptions listenOptions, ServerOptionsSelectionCallback serverOptionsSelectionCallback, object state, TimeSpan handshakeTimeout)`

Author: @Rick-Anderson can I xref to specific implementations of a method?

Reviewer: Yes, but you usually use the [Docs Authoring Pack](https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-authoring-pack)

Feel free to update an issue or create a new issue and assign to me.

Author: https://github.com/dotnet/AspNetCore.Docs/issues/23432
